benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/JavaClientBenchmarkRunner.java

```diff
@@ -0,0 +1,256 @@
+      PGAdapterConfiguration pgAdapterConfiguration,
+      SpannerConfiguration spannerConfiguration,
+      Metrics metrics) {
+    super(statistics, tpccConfiguration, pgAdapterConfiguration, spannerConfiguration, metrics);
+  }
+
+  void setup() throws SQLException, IOException {
+    this.spanner = createSpanner();
+  }
+
+  void teardown() throws SQLException {
+    spanner.close();
+  }
+
+  void executeStatement(String sql) throws SQLException {
+    if (sql.equals("begin transaction")) {
+      DatabaseClient databaseClient =
+          spanner.getDatabaseClient(
+              DatabaseId.of(
+                  spannerConfiguration.getProject(),
+                  spannerConfiguration.getInstance(),
+                  spannerConfiguration.getDatabase()));
+      transactionManagerThreadLocal.set(databaseClient.transactionManager());
+      transactionThreadLocal.set(transactionManagerThreadLocal.get().begin());
+    } else if (sql.equals("commit")) {
+      try {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        transactionManagerThreadLocal.get().commit();
+        Duration executionDuration = stopwatch.elapsed();
+        metrics.recordLatency(executionDuration.toMillis());
+      } catch (AbortedException e) {
+        // Ignore the aborted exception. Roll back the transaction.
+        transactionManagerThreadLocal.get().rollback();
+      } finally {
+        transactionManagerThreadLocal.get().close();
+      }
+    } else if (sql.startsWith("rollback")) {
+      transactionManagerThreadLocal.get().rollback();
+      transactionManagerThreadLocal.get().close();
+    } else if (sql.equals("set transaction read only")) {
+      // No-op because TransactionManager only supports read/write transactions.
+    }
+  }
+
+  void bindParams(Statement.Builder builder, Object[] params) throws SQLException {
+    for (int i = 0; i < params.length; i++) {
+      Object param = params[i];
+      if (param instanceof BigDecimal) {
+        builder.bind("p" + (i + 1)).to(Value.pgNumeric(((BigDecimal) param).toPlainString()));
+      } else if (param instanceof String) {
+        builder.bind("p" + (i + 1)).to((String) param);
+      } else if (param instanceof Integer) {
+        builder.bind("p" + (i + 1)).to((int) param);
+      } else if (param instanceof Long) {
+        builder.bind("p" + (i + 1)).to((long) param);
+      } else {
+        throw new SQLException(String.format("Unknown type for the parameter: %s", param));
+      }
+    }
+  }
+
+  Object[] paramQueryRow(String sql, Object[] params) throws SQLException {
+    ParametersInfo parametersInfo = convertPositionalParametersToNamedParameters(sql, '?', "$");
+    Statement.Builder builder = Statement.newBuilder(parametersInfo.sqlWithNamedParameters);
+    bindParams(builder, params);
+    Statement stmt = builder.build();
+
+    ResultSet resultSet = transactionThreadLocal.get().executeQuery(stmt);
+    Stopwatch stopwatch = Stopwatch.createStarted();
```

Comment:
nit: If you want to have a fully apples-to-apples comparison, then this should be moved to before the `executeQuery(..)` call above. `executeQuery(..)` does not actually execute the query, but it can include checking out the session from the pool. This is also always included in the `executeQuery(..)` methods in the JDBC driver and PGAdapter.

Reply:
Good catch. Fixed.